tests/PHPStan/Rules/Properties/data/require-implements.php

```diff
@@ -0,0 +1,37 @@
+<?php declare(strict_types=1);
+
+namespace RequireImplements;
+
+interface MyInterface
+{
+	public function doSomething(): string;
+
+	static public function doSomethingStatic(): int;
+}
+
+/**
+ * @phpstan-require-implements MyInterface
```

Comment:
in this case here, the userland code needs to be

```
class MyClass implements MyInterface
{
	use MyTrait;
}
```

which means for `require-implements` I don't need a class-reflection extension, because the default-extension will already allow access to the interface methods, right?

we only need additional rules to verify the `@phpstan-require-implements MyInterface` is used properly, right?

Reply:
turned differently: my thinking is we need the class-reflection extension only to allow access to properties/methods for the `@phpstan-require-extends` case?

Reply:
I still think we only need the class-reflection extension for the case where we use `@phpstan-require-extends` on a interface.

when using `@phpstan-require-extends` or `@phpstan-require-implements` on a trait, we only need to create a few rules to sanity check the annotations context and params are valid

Reply:
Yeah, looks like it 👍